core/templates/base-components/oppia-footer.component.ts

```diff
@@ -37,43 +37,64 @@ import './oppia-footer.component.css';
   templateUrl: './oppia-footer.component.html',
   styleUrls: ['./oppia-footer.component.css'],
 })
-export class OppiaFooterComponent {
+export class OppiaFooterComponent implements OnDestroy {
   emailAddress: string | null = null;
   name: string | null = null;
   siteFeedbackFormUrl: string = AppConstants.SITE_FEEDBACK_FORM_URL;
   currentYear: number = new Date().getFullYear();
   PAGES_REGISTERED_WITH_FRONTEND = AppConstants.PAGES_REGISTERED_WITH_FRONTEND;
-
   BRANCH_NAME = AppConstants.BRANCH_NAME;
-
   SHORT_COMMIT_HASH = AppConstants.SHORT_COMMIT_HASH;
 
+  private debounceSubject: Subject<void> = new Subject();
+  private unsubscribe$: Subject<void> = new Subject();
   versionInformationIsShown: boolean =
     this.router.url === '/about' && !AppConstants.DEV_MODE;
 
+  isSubmitting: boolean = false;
```

Comment:
We try to avoid starting variables that denote nouns with verbs ("is" is a verb). Try something like submissionIsInProgress, perhaps.

Reply:
> We try to avoid starting variables that denote nouns with verbs ("is" is a verb). Try something like submissionIsInProgress, perhaps.

Got it, thanks! I'll update it to submissionIsInProgress for better clarity.